domain-model-assistant/Assets/Components/Scripts/CompartmentedRectangle.cs

```diff
@@ -214,38 +217,54 @@ public override List<Vector2> GetConnectionPointsLocations()
 
     public GameObject GetPopUpMenu()
     {
-        return this.popupMenu;
+        return popupMenu;
     }
     
     public void SetLine()
     {
-        Debug.Log("set line heard");
-        _diagram.gameObject.GetComponent<DrawLine>().AddCompartmentedRectangle(this.gameObject);
-        this.popupMenu.GetComponent<PopupMenu>().Close(); // close the popup menu
+        Debug.Log("CompartmentedRectangle.SetLine() called");
+        _diagram.gameObject.GetComponent<DrawLine>().AddCompartmentedRectangle(gameObject);
+        popupMenu.GetComponent<PopupMenu>().Close(); // close the popup menu
     }
 
-    public void toggleHighlight(){
-        if(isHighlighted){
-            isHighlighted = false;
-            this.transform.GetChild(1).gameObject.GetComponent<Image>().color = headerColor;
-            this.transform.GetChild(3).gameObject.GetComponent<Image>().color = sectionColor;
-            this.transform.GetChild(4).gameObject.GetComponent<Image>().color = sectionColor;
-            //restore color
+    public void ToggleHighlight()
+    {
```

Comment:
We need to decide what "highlight" means, since this word can refer to highlighting a class which is selected in light blue, or highlighting a class with a mistake in yellow. We should also figure out how to highlight a class that is both wrong and selected by the user